img2pdf: add page (#6519)
tldr-pages/tldr: pages/common/img2pdf.md

# img2pdf

> Command-line lossless converter of raster images to PDF.
> More information: <https://gitlab.mister-muffin.de/josch/img2pdf>.

> - Convert multiple images to PDF, each image being on its own page:

Reviewer: ```suggestion
- Convert multiple images to PDF, each image being on its own document:
```
```suggestion
- Convert multiple images to PDF, each image being on its own PDF:
```

Author: Actually, it ouputs them to a single file. This PDF contains multiples pages, one per input.

Reviewer: :+1:

`img2pdf {{path/to/image1.jpg}} {{path/to/image2.jpg}} --output {{path/to/file.pdf}}`

- Convert all PNGs in the current directory to a single PDF:

`img2pdf {{*.png}} --output {{path/to/file.pdf}}`

- Convert only the first frame of a multi-frame image to PDF:

`img2pdf {{path/to/file.gif}} --first-frame-only --output {{path/to/file.pdf}}`

- Auto orient the image, use a page size of A4 in landscape mode, and set a border of 2cm horizontally and 5.1cm vertically:

> `img2pdf {{path/to/file.jpg}} --auto-orient --pagesize {{A4^T}} --border {{2cm:5.1cm}} --output {{path/to/file.pdf}}`

Reviewer 2: ```suggestion
`img2pdf {{path/to/file.jpg}} --auto-orient --pagesize {{A4^T}} --border {{2cm}}:{{5.1cm}} --output {{path/to/file.pdf}}`
```

- Shrink only larger images to a 10cm by 15cm rectangle inside a 30x20cm page:

> `img2pdf {{path/to/file.tiff}} --pagesize {{30cmx20cm}} --imgsize {{10cmx15cm}} --fit {{shrink}} --output {{path/to/file.pdf}}`

Reviewer 2: I think this would improve readability
```suggestion
`img2pdf {{path/to/file.tiff}} --pagesize {{30cm}}x{{20cm}} --imgsize {{10cm}}x{{15cm}} --fit {{shrink}} --output {{path/to/file.pdf}}`
```

Reviewer 3: Good idea!

# img2pdf

> Command-line lossless converter of raster images to PDF.
> More information: <https://gitlab.mister-muffin.de/josch/img2pdf>.

- Convert images to PDF:

`img2pdf {{path/to/image1.jpg}} {{path/to/image2.jpg}} --output {{path/to/file.pdf}}`

- Convert all PNGs in the current directory to a single pdf:

> `img2pdf *.{{png}} -o out.pdf`

Reviewer 4: ```suggestion
`img2pdf {{*.png}} --output {{path/to/file.pdf}}`
```

> - Convert only the first page of multi-page images to pdf:

Reviewer 3: Perhaps this will read better?
```suggestion
- Convert only the first frame of multi-frame images to a pdf:
```

> - Convert only the first frame of a multi-frame image to a pdf:

Reviewer 2: ```suggestion
- Convert only the first frame of a multi-frame image to PDF:
```

`img2pdf {{path/to/file.gif}} --first-frame-only --output {{path/to/file.pdf}}`

> - Auto orient to A4 page size in landscape mode and a border of 2cm horizontally and 5.1cm vertically:

Reviewer 3: > 
Perhaps this will read better?
```suggestion
- Auto orient image, use a page size of A4 in landscape mode, and set a border of 2cm horizontally and 5.1cm vertically:
```

`img2pdf {{path/to/file.jpg}} --auto-orient --pagesize {{A4^T}} --border {{2cm:5.1cm}} --output {{path/to/file.pdf}}`

- Shrink only larger images to a 10cm by 15cm rectangle inside a 30x20cm page:

> `img2pdf example.tiff --pagesize {{30cmx20cm}} --imgsize {{10cmx15cm}} --fit {{shrink}} -o out.pdf`

Reviewer 4: ```suggestion
`img2pdf {{path/to/file.tiff}} --pagesize {{30cmx20cm}} --imgsize {{10cmx15cm}} --fit {{shrink}} --output {{path/to/file.pdf}}`
```

> - Set metadata:

Reviewer 4: Could be this better?
```suggestion
- Convert an image to PDF including specific metadata:
```

> - Convert an image to PDF including specific metadata:

Reviewer 3: Note that images can contain metadata in the form of EXIF/XMP tags, so it' worth differentiating here. Perhaps this will read better?
```suggestion
- Convert an image to PDF, and specify metadata for the resulting file:
```

> `img2pdf example.png --title {{title}} --author {{author}} --creationdate {{1970-01-31}} --keywords {{keywords}} --subject {{subject}} -o out.pdf`

Reviewer 4: ```suggestion
`img2pdf {{path/to/file.png}} --title {{title}} --author {{author}} --creationdate {{1970-01-31}} --keywords {{keywords}} --subject {{subject}} --output {{path/to/file.pdf}}`
```

> `img2pdf {{path/to/file.png}} --title {{title}} --author {{author}} --creationdate {{1970-01-31}} --keywords {{keywords}} --subject {{subject}} --output {{path/to/file.pdf}}`

Reviewer 5: Since `--keywords` accepts multiple arguments, I would have to guess the separator. You should indicate it by including multiple keywords. (`keyword1,keyword2` if the separator is `,`)

Reviewer 3: Good point. What about this?
```suggestion
`img2pdf {{path/to/file.png}} --title {{title}} --author {{author}} --creationdate {{1970-01-31}} --keywords {{keyword1,keyword2,...}} --subject {{subject}} --output {{path/to/file.pdf}}`
```

Author: you can separate them by a space `--keywords keyword1 keyword2`. These keywords don't show up with Dolphin File Manager, only with `exiftools` and Okular (PDF viewer). Maybe a bug? I don't know.

Reviewer 2: I think we should keep this example rather simple. Also the flag name kind of indicates that it is at least possible to give multiple argument (it's `keywords` and not `keyword`). For more information we still have man pages. 

But if it makes sense sense to write `--keywords {{keyword1 keyword2 ...}}` I see no problem with it.

> - Force RGB as colorspace:

Reviewer 3: > 
Perhaps this will read better?
```suggestion
- Force the colorspace to RGB:
```

> - Force RGB as colorspace:

Reviewer 5: To me this description is a bit unclear. If I have a YUV file, does it convert YUV to RGB? Does it read my YUV image as RGB?

For the first one, you could use something like `Convert an image to PDF with the RGB colorspace`

Author: After a few tests, I couldn't figure out what this did in a practice. If I used the same colorspace as the input image, then no change. In contrast, other options would completely mess with the image. I assume it treats the file as a specific colorspace despite what the metadata may say. I could remove this line, doesn't seem necessary for a TLDR.

Reviewer 2: Sounds good
